packages/davinci-client/src/lib/wellknown.api.ts

```diff
@@ -0,0 +1,27 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query';
+import { Endpoints, OpenIdResponse } from './wellknown.types';
+
+export const wellknownApi = createApi({
+  reducerPath: 'wellknown',
+  baseQuery: fetchBaseQuery({
+    prepareHeaders: (headers) => {
+      headers.set('Accept', 'application/json');
+      return headers;
+    },
+  }),
+  endpoints: (builder) => ({
+    wellknown: builder.query<Endpoints, string>({
+      query: (endpoint: string) => ({ url: endpoint }),
+
+      transformResponse: (value: OpenIdResponse): Endpoints => {
+        return {
+          authorize: value.authorization_endpoint,
+          issuer: value.issuer,
+          introspection: value.introspection_endpoint,
+          tokens: value.token_endpoint,
+          userinfo: value.userinfo_endpoint,
+        };
+      },
```

Comment:
If we are going to store the endpoints in the `config` slice, I think I'd prefer to keep the response unmodified here. That way it's cached untouched, and the `config` slice's reducer maps the original response properties appropriately. This is what we do in the DaVinci API.

Reply:
What do you mean by "its cache is untouched" The `transformResponse` occurs before caching is what I understand?

Reply:
If you transform the response as you are doing here, you are going to cache the transformation result, not the original response from the server. I'd personally prefer to cache the original.

![CleanShot 2024-11-19 at 13 21 00](https://github.com/user-attachments/assets/d38c2bb4-740d-4881-b88a-0a7ca873a24e)

Reply:
So at what stage would you rather the transformation happened? 

- Would we pass the entire openId response into the reducer and have the reducer pull out the endpoints?
- Would we do it ourselves before calling `set` action? 

Its a minor thing i'm just unsure where the best place to do this is if not in `transformResponse`. Doesn't matter much to me either way

Reply:
I'd say just pass the original response to the config slice. You don't have to transform or normalize the response from the wellknown endpoint since it follows a spec. All responses from the wellknown should be the same. That means you can just pass the original response to the config slice and have it map what it needs to the stored config.